docs/rules/prefer-explicit-assert.md

```diff
@@ -0,0 +1,77 @@
+# Suggest using explicit assertions rather than just `getBy*` queries (prefer-explicit-assert)
+
+Testing Library `getBy*` queries throw an error if the element is not
+found. Some users like this behavior to use the query itself as an
+assert for the element existence in their tests, but other users don't
+and prefer to explicitly assert the element existence, so this rule is
+for users from the latter.
+
+## Rule Details
+
+This rule aims to encourage users to explicitly assert existence of
+elements in their tests rather than just use `getBy*` queries and expect
+it doesn't throw an error so it's easier to understand what's the
+expected behavior within the test.
+
+> ⚠️ Please note that this rule is recommended to be used together with
+> [prefer-expect-query-by](docs/rules/prefer-expect-query-by.md), so the
+> combination of these two rules will force users to do 2 actual
+> changes: wrap `getBy*` with `expect` assertion and then use `queryBy*`
+> instead of `getBy*` for asserting.
+
+Examples of **incorrect** code for this rule:
+
+```js
+// just calling `getBy*` query expecting not to throw an error as an
+// assert-like method, without actually either using the returned element
+// or explicitly asserting
+getByText('foo');
+```
+
+Examples of **correct** code for this rule:
+
+```js
+// wrapping the get query within a `expect` and use some matcher for
+// making the assertion more explicit
+expect(getByText('foo')).toBeDefined();
+
+// ⚠️ `getBy*` should be replaced by `queryBy*` when combined with `prefer-expect-query-by` rule
+expect(queryByText('foo')).toBeDefined();
+
+// even more explicit if you use `@testing-library/jest-dom` matcher
+// for checking the element is present in the document
+expect(queryByText('foo')).toBeInTheDocument();
+
+// Doing something with the element returned without asserting is absolutely fine
+await waitForElement(() => getByText('foo'));
+fireEvent.click(getByText('bar'));
+const quxElement = getByText('qux');
+
+// call directly something different than Testing Library query
+getByNonTestingLibraryVariant('foo');
+```
+
+## Options
+
+This rule accepts a single options argument:
+
+- `customQueries`: this array option allows to extend default Testing
```

Comment:
Nit: maybe `customQueryNames`?

Reply:
That was actually my first approach and then I renamed it, so I'm happy to go back to that one if it feels the right name.

Reply:
Ah ok, I think it's easier to understand if the name is more specific. I don't have a strong opinion though, up to you to decide.

Reply:
Done, I agree with you in having it more specific.